etcdserver/server.go

```diff
@@ -632,6 +628,21 @@ func (s *EtcdServer) Lead() uint64 { return atomic.LoadUint64(&s.r.lead) }
 
 func (s *EtcdServer) Leader() types.ID { return types.ID(s.Lead()) }
 
+// RequestTimeout returns an appropriate timeout to wait for request to be committed.
+func (s *EtcdServer) RequestTimeout() time.Duration {
+	// expectedCommitDelay
+	// + 2 * heartbeat for possible message lost
+	// + 2 * election timeout for possible leader election
+	return s.expectedCommitDelay() + 2*time.Duration(s.cfg.TickMs)*time.Millisecond + 2*time.Duration(s.cfg.ElectionTicks)*time.Duration(s.cfg.TickMs)*time.Millisecond
+}
+
+func (s *EtcdServer) expectedCommitDelay() time.Duration {
```

Comment:
It needs a better naming definitely. 

I want to mention the upper limit of practical time between follower doing proposal and follower finding out the proposal has been committed in good case.

For message transfer, it includes 1. follower sends proposal to leader 2. leader sends msgApp to follower 3. follower sends msgAppResp to leader 4. leader announce committed to follower. This is 2 RTT.

For fixed time, it needs to include computation and queue time. So i set 5s for it.